configure.ac

```diff
@@ -287,6 +287,15 @@ DX_MAN_FEATURE(ON)
 DX_PDF_FEATURE(ON)
 DX_INIT_DOXYGEN([UCX],[doc/doxygen/ucxdox],[doc/doxygen-doc])
 
+#
+#Check doxygen initialization status
+#
+AS_IF([test "x$with_docs_only" == xyes],
+      [AS_IF([DX_TEST_FEATURE(doc) && DX_TEST_FEATURE(html) &&
+              DX_TEST_FEATURE(man) && DX_TEST_FEATURE(pdf)],
+             [AS_MESSAGE([==============================])],
+             [AC_MSG_ERROR([--with-only-docs was specified, but unable to configure it])])])
```

Comment:
maybe mention doxygen explicitly as a hint what is missing?

Reply:
This tests doxygen, pdflatex and tools for html, man that we're using for documentation.
`DX_INIT_DOXYGEN` macro reports warnings when it is unable to find requested tools or requested version of tools.

e.g.:
```
checking for doxygen... /bin/doxygen
configure: WARNING: doxygen version 1.8.5 is bad. Required version: 1.8.6 and above
checking for perl... /bin/perl
configure: error: --with-only-docs was specified, but unable to configure it
```
```
configure: WARNING: pdflatex not found - will not generate doxygen PDF documentation
checking for makeindex... no
checking for bibtex... no
checking for egrep... /bin/egrep
configure: error: --with-only-docs was specified, but unable to configure it
```

WDYT? Is it enough?

Reply:
aa, ok then

Reply:
1. we can move this check under the already-existing `AS_IF([test "x$with_docs_only" == xyes],` on line 134
1. why need to print `=====`?
1. the error message should be `configure: error: --with-only-docs was specified, but doxygen is not found`
1. no need to fail if pdf is not available. what if html doc is good enough for us? then --with-doc-only could not be used?

Reply:
> 2\. why need to print `=====`?

I forgot to remove this
was added for debug needs, there is no needs anymore

> 4\. no need to fail if pdf is not available. what if html doc is good enough for us? then --with-doc-only could not be used?

Probably, we need `--enable-{ pdf, html, man }-docs`
1. `--with-docs-only` fails if there is not doxygen
2. If `--enable-{ pdf, html, man }-docs` specified, it enables strict mode (i.e. if some tools is not found, ./configure fails)

Reply:
WDYT about approach above?

Reply:
> 1. we can move this check under the already-existing `AS_IF([test "x$with_docs_only" == xyes],` on line 134

we should do these checks after `DX_INIT_DOXYGEN` macro, because `DX_INIT_DOXYGEN` inititalizes values that are tested inside `DX_TEST_FEATURE(...)` macro

Reply:
1. Sp, need to move doxygen checks before line 134
2. So if user specifies `--with-docs-only --disable-pdf-docs` configure will fail?

Reply:
> 2\. So if user specifies `--with-docs-only --disable-pdf-docs` configure will fail?

No, it will succeed. Also, it will succeed if just specified `--with-docs-only`

But it user wants pdf (`--with-docs-only --enable-pdf-docs` specified), it will fail if pdflatex is unavailable on the system

Reply:
ok

Reply:
> ok

done